common/djangoapps/student/views.py

```diff
@@ -712,6 +714,58 @@ def dashboard(request):
     return render_to_response('dashboard.html', context)
 
 
+@login_required
+@ensure_csrf_cookie
+def revisions(request):
+    """
+    Return a JSON list of all revisions for a user. Each revision includes a name, due date, and URL.
+    """
+    user = request.user
+
+    revisions = _get_pending_revisions(user)
+    json_revisions = json.dumps(revisions)
+
+    return HttpResponse(json_revisions)
+
+
+def _get_pending_revisions(user):
+    """
+    Returns information about each problem that needs revision for a given user, including a display name, the due date
+    of the revision, and a courseware URL.
+    """
+    # TODO: Fetch this list of serialized locators from the Domoscio API.
+    serialized_locators = [
+        'block-v1:asdf+asdf+asdf+type@problem+block@9adc878ed55943caa7d4b4cd48e52752',
+        'block-v1:asdf+asdf+asdf+type@problem+block@0c4c0798cf3847ae9c706cdcb9442c62',
+    ]
+
+    locators = map(BlockUsageLocator.from_string, serialized_locators)
+
+    revision_data = []
+    for locator in locators:
+        # Resolve the courseware URL for the revision xblock.
+        (
+            course_key, chapter, section, vertical_unused,
+            position, final_target_id
+        ) = path_to_location(modulestore(), locator)
+
+        url = reverse(
+            'courseware_position',
+            args=(unicode(course_key), chapter, section, navigation_index(position))
+        )
+
+        xblock = modulestore().get_item(locator)
+        # TODO: Use either xblock fields or values returned from the Domoscio API to populate the displayed
+        #       name and date; examples are listed as line comments below.
```

Comment:
The xblock type is in a field (don't remember the name). If we grab all of the Adaptive Content Blocks, I think we can supply the block name as a qualifier in a modulestore fetch (like the [discussions XBlock](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/django_comment_client/utils.py#L133-L138) does), and then filter by the current user.

I'll leave this for OC-2154, as the exact implementation will require a little tinkering and will likely not be precisely what I'm saying here. :)